Automattic/simplenote-macos
# Charlie/passkey registration

### Fix
We are adding passkey support to SNMac!  This will come in a couple of stages, first in this PR we are adding the ability to register a passkey on your account.  Signing in with a passkey will come later.

With a signed in account, from the preferences view there is a new button "Add passkey sign in" tapping that will allow a user to authorize there account and create a new passkey in their iCloud Keychain.  Once added they should be able to use this passkey to sign into simplenote on Mac or iOS with the same passkey as long as that version of simplenote is updated with the passkey auth info.

### Test
1. Sign in to a simplenote account and open the preferences view
2. tap on the Add Passkey Sign in
3. Confirm that a blocking spinner appears while we fetch a challenge
4. If successful you should see an apple auth appear.  Dismiss it without verifying.  Confirm you see a failure alert and the spinner disappears
5. Tapp on the add passkey button again and this time when the apple auth appears continue to add a passkey.  When it is complete you should see a success message.  
6. Go to the passkey db and confirm your new passkey was added (ping me for a link if need be)

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:

> These changes do not require release notes.
